### PR TITLE
softmax: enable half_to_float

### PR DESCRIPTION
**Pull request by weishi-deng**
To enable half_to_float flag in softmax.

### CODE REVIEW

**Comment by xytintel** [comment]
Is there any test case?

**Comment by weishi-deng** [comment]
> Is there any test case?

UT updated.

**Review by xytintel — APPROVED** [review]

File: test/regressions/test_softmax.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                x = torch.randn(shape[i]).to(input_type)
+                grad = torch.randn(shape[i]).to(output_type)
+                x_cpu = x.clone().requires_grad_()
+                y_cpu = F.softmax(x_cpu, dim, dtype=output_type)
+                y_cpu.backward(grad.clone())
+
+                x_xpu = x.clone().to(xpu_device).requires_grad_()
+                y_xpu = F.softmax(x_xpu, dim, dtype=output_type)
+                y_xpu.backward(grad.clone().to(xpu_device))
+                self.assertEqual(y_cpu, y_xpu.cpu())
```
Comment (guangyey) [inline test/regressions/test_softmax.py:37]: I have a question, without https://github.com/pytorch/pytorch/pull/150278, why could this UT pass.

Comment (weishi-deng) [reply test/regressions/test_softmax.py:37]: > I have a question, without [pytorch/pytorch#150278](https://github.com/pytorch/pytorch/pull/150278), why could this UT pass.

I think this ut doesn't work as pytorch do not dispatch to this path. It can pass my local test. So the next step is to update the torch-xpu-ops commit and test this ut in pytorch, I think.

Comment (guangyey) [reply test/regressions/test_softmax.py:37]: Make sense. Thanks.
